dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -69,22 +77,73 @@
             .setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
             .findAndRegisterModules();
 
-    private AppSecrets readJson(final char[] chars) throws DotDataException {
+    /**
+     * Given the AppSecrets this will return a char array representing the deserialized json object.
+     * No strings are created in the transformation process.
+     * @param object
+     * @return
+     * @throws DotDataException
+     */
+    @VisibleForTesting
+    char[] toJsonAsChars(final AppSecrets object) throws DotDataException {
         try {
-            return jsonMapper.readValue(new String(chars), AppSecrets.class);
+            final byte [] bytes = jsonMapper.writeValueAsBytes(object);
+            return bytesToCharArrayUTF(bytes);
         } catch (IOException e) {
             throw new DotDataException(e);
         }
     }
 
-    private String toJsonAsString(final AppSecrets object) throws DotDataException {
+    /**
+     * Takes a char array representation of a json secret and generates the domain object.
+     * No strings are created in the transformation process.
+     * @param chars
+     * @return The Secrets domain model.
+     * @throws DotDataException
+     */
+    @VisibleForTesting
+    AppSecrets readJson(final char[] chars) throws DotDataException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)